src/slopes.jl

```diff
@@ -0,0 +1,198 @@
+# Reference : Dietmar Ratz - An Optimized Interval Slope Arithmetic and its Application
+using IntervalArithmetic, ForwardDiff
+import Base: +, -, *, /, ^, sqrt, exp, log, sin, cos, tan, asin, acos, atan
+import IntervalArithmetic: mid, interval
+
+function slope{T}(f::Function, x::Interval{T}, c::T)
+    try
+        f(slope_var(x, c)).s
+    catch y
+        if isa(y, MethodError)
+            ForwardDiff.derivative(f, x)
+        end
+    end
+end
+
+struct Slope{T}
+    x::Interval{T}
+    c::Interval{T}
+    s::Interval{T}
+end
+
+Slope{T}(c) where T = Slope{T}(c, c, 0)
+
+function slope_var{T}(v::T)
```

Comment:
Please use the new syntax with where instead of this syntax everywhere for parameterised functions.

Reply:
Oh okay, I was using the syntax that was prevalent in the repo already. I will change accordingly.